### ISSUE TITLE
Binary operators and broadcast syntax don't interact well

### ISSUE THREAD

**Issue by fcard** [JuliaLang/julia#16993]
Currently, it's necessary to wrap binary operators in parentheses to use them with the new broadcast syntax. Otherwise:

``` julia
julia> *.(2,[1,2,3])
ERROR: syntax: "*" is not a unary operator
 in eval(::Module, ::Any) at ./boot.jl:231
 in macro expansion at ./REPL.jl:92 [inlined]
 in (::Base.REPL.##1#2{Base.REPL.REPLBackend})() at ./event.jl:46
```

or

``` julia
julia> +.(1,[1,2,3])
ERROR: syntax: numeric constant "." cannot be implicitly multiplied because it ends with "."
 in eval(::Module, ::Any) at ./boot.jl:231
 in macro expansion at ./REPL.jl:92 [inlined]
 in (::Base.REPL.##1#2{Base.REPL.REPLBackend})() at ./event.jl:46
```

P.S. Also affects the not nearly as important but still _happenable_ cases of using unicode operators as normal variables:

``` julia
type GreekTemple
  number_of_columns::Int
end
const ⟰ = GreekTemple(4)

println(⟰.number_of_collumns) 
# My attempt to build a compilable greek city with unicode art is ruined >:

type Sadness
  sadness_rating::Int
end
const >: = Sadness(10000)

println(>:.sadness_rating) # >: >: >: >: >: >:
```

edit: I was so used calling custom binary operators unicode that I forgot that some of them are ascii...

**Comment by JeffBezanson** [comment]
The case with the "implicitly multiplied" error looks like a bug; there is no numeric constant present so at the very least it needs a better error.

I don't think we ever intended to allow accessing fields of operators with `.`, but maybe we can. It's a bit ambiguous with e.g. `+.2`, which is unary + applied to 0.2.

**Comment by fcard** [comment]
Well, `a.2` is invalid syntax as well, so `+.2` could still mean what it does now. And accessing fields of operators already works with `()`, as in `(⟰).number_of_collumns`, so that would just make it easier for when it's necessary... although, I don't see myself personally using that, so I am fine either way, just reportin'!

Also, I just noticed that unary operators other than `+` and `-` are giving a different error:
```julia
julia> !.(1)
ERROR: syntax: invalid identifier name "."
 in eval(::Module, ::Any) at ./boot.jl:231
 in macro expansion at ./REPL.jl:92 [inlined]
 in (::Base.REPL.##1#2{Base.REPL.REPLBackend})() at ./event.jl:46
```
I am guessing because the parser is trying to parse `.(1)` separately.
(and `-` gives `numeric constant ".-"` instead of just ".")

**Comment by stevengj** [comment]
Note that this should be mostly addressed in 0.6 by #17623, since you will just do `2 .+ [1,2,3]` to get a fused broadcast.

**Comment by stevengj** [comment]
Closed by 17623.
